# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1335546679 14400
#      Fri Apr 27 13:11:19 2012 -0400
# Node ID 721442e1b82cd6f7e78597a17ce8116d2a7665d4
# Parent  4dc85c4f151b35e630bd1ad65896fcbb32e7b0e8
For gnuplot, render the legend's text objects using the specified textcolor.

legend.m: Modify demo.
__go_draw_axes__.m: Respect the "textcolor" property for the legend.

diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -969,16 +969,17 @@ function updateline (h, d, hlegend, line
 endfunction
 
 %!demo
 %! plot (rand (2,2)) ;
 %! h = legend ('a', 'b') ;
 %! legend ('right') ;
 %! set (h, 'textposition', 'left')
 %! set (h, 'textposition', 'right')
+%! set (h, 'textcolor', [1 0 1])
 
 %!demo
 %! clf;
 %! x = 0:1;
 %! plot (x,x,';I am Blue;', x,2*x,';I am Green;', x,3*x,';I am Red;');
 
 %!demo
 %! clf;
diff --git a/scripts/plot/private/__go_draw_axes__.m b/scripts/plot/private/__go_draw_axes__.m
--- a/scripts/plot/private/__go_draw_axes__.m
+++ b/scripts/plot/private/__go_draw_axes__.m
@@ -1576,18 +1576,19 @@ function __go_draw_axes__ (h, plot_strea
           pos = "";
       endswitch
       if (__gnuplot_has_feature__ ("key_has_font_properties"))
         [fontname, fontsize] = get_fontname_and_size (hlgnd);
         fontspec = create_fontspec (fontname, fontsize, gnuplot_term);
       else
         fontspec = "";
       endif
-      fprintf (plot_stream, "set key %s %s;\nset key %s %s %s %s;\n",
-               inout, pos, box, reverse, horzvert, fontspec);
+      colorspec = get_text_colorspec (hlgnd.textcolor, mono);
+      fprintf (plot_stream, "set key %s %s;\nset key %s %s %s %s %s;\n",
+               inout, pos, box, reverse, horzvert, fontspec, colorspec);
     else
       fputs (plot_stream, "unset key;\n");
     endif
     fputs (plot_stream, "set style data lines;\n");
 
     cmap = [cmap; addedcmap];
     cmap_sz = cmap_sz + rows(addedcmap);
     if (length(cmap) > 0)
